### ISSUE TITLE
"Collision" component doesn't know when entity resizes

### ISSUE THREAD

**Issue by starwed** [craftyjs/Crafty#448]
Adding a collision component seems to prevent mouse events from being registered: 

[jsfiddle example](http://jsfiddle.net/jLzJm/3/)

_e:_ Turns out this is because "Collision" doesn't know when an entity resizes.  I propose to make Collision match an objects dimensions until a different polygon is explicitly set.

**Comment by sbyrnes321** [comment]
First, notice that the jsfiddle puts the collision box (`this.map`) at width=0 and length=0. That's because the collision component is added before the height and width are set. Then the connection to mouse is in the isAt function:

```
isAt: function (x, y) {
    if (this.mapArea) {
        return this.mapArea.containsPoint(x, y);
    } else if (this.map) {
        return this.map.containsPoint(x, y);
    }
    return this.x <= x && this.x + this.w >= x &&
           this.y <= y && this.y + this.h >= y;
```

`mapArea` is defined when you use an `areaMap` from the `Mouse` component, while `map` is an analogous thing for the `Collision` component. I guess the assumption is that if you have a specific area for collisions, the same shape should be the default for detecting mouse-clicks.

So in the particular case of this jsfiddle, you could make it work by running `.collision()` after the height and width are set, which resets the collision box to the size of the rectangle. Or, don't add the component in the first place, until after the `attr` command that sets height and width.

I'm just describing what's happening. It's not obvious to me whether we should change the behavior or add to the documentation or leave it as is.

**Comment by starwed** [comment]
Ah, I've run into that before in other contexts, didn't think about it here.

It's actually _really_ annoying that "Collision" doesn't naturally track the size of the object.  Maybe calling collision() could add a listener for "Change" events and track the width and height of the entity.  If ever that's overridden by handing explicit parameters to collision, it could remove the listener.  I think in almost all cases this is what people really want from "Collision".

**Comment by starwed** [comment]
I have a mostly working implementation of a responsive collision hitbox; it required adding a "Resize" event when h or w changes.

There's a couple of bugs I need to work out (rotations complicate things) but I wanted to give a heads up that I was working on this.

**Comment by starwed** [comment]
Fixed by #485
